### PR TITLE
octave: fixes for Qt 6.7.0

### LINKED ISSUE #170616: Octave 9.1.0 opengl bug interacting with qt6-6.7.0
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
I do not use homebrew.  This is an attempt at getting an upstream patch in to fix crashes for homebrew users.
```

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I can't seem to get your system to accept my patch.  Here is my best go at it.  In the following code block the patch to octave.rc should (I hope) download a patch from savannah.gnu.org and apply it to your source tree prior to build.  This should fix the bug with octave 9.1.0 interacting with qt6-6.7.0.

```
diff --git a/Formula/o/octave.rb b/Formula/o/octave.rb
index 60f8aea563d..c11a87df330 100644
--- a/Formula/o/octave.rb
+++ b/Formula/o/octave.rb
@@ -81,6 +81,12 @@ class Octave < Formula
 
   fails_with gcc: "5"
 
+  # Fix opengl-partial-update bug causing crashes on figure() and plot().
+  patch do
+    url "https://file.savannah.gnu.org/file/bug65605-qt6-opengl-partial-update.patch?file_id=55978"
+    sha256 "4f258a90dcb6b5d7f1af466dfc2b2fd6413968070ab224e4b368d355ec257ca9"
+  end
+
   def install
     # Default configuration passes all linker flags to mkoctfile, to be
     # inserted into every oct/mex build. This is unnecessary and can cause
```

### What happened (include all command output)?

I got a report from a homebrew user in the octave mailing lists saying his build crashes.

### What did you expect to happen?

Opening octave and running figure() will cause a crash.  With this patch applied a figure window will open instead.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
No instructions here.  Just trying to help out.
```

### PR DESCRIPTION
**Pull request by fxcoudert**
Fixes https://github.com/Homebrew/homebrew-core/issues/170616

### CODE REVIEW

**Comment by chenrui333** [comment]
duplicate of https://github.com/Homebrew/homebrew-core/pull/170628?

**Comment by fxcoudert** [comment]
I didn't see https://github.com/Homebrew/homebrew-core/pull/170628
But it only has part of the fix

**Review by carlocab — APPROVED** [review]

**Comment by chenrui333** [comment]
> I didn't see #170628 But it only has part of the fix

all good 👍

**Comment by fxcoudert** [comment]
`dynare` test failure is due to https://github.com/Homebrew/brew/issues/17203

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

File: Formula/o/octave.rb [hunk 1]
```diff
@@ -35,7 +36,10 @@ class Octave < Formula
   end
 
   # Complete list of dependencies at https://wiki.octave.org/Building
+  depends_on "autoconf" => :build # for the patches
+  depends_on "automake" => :build # for the patches
   depends_on "gnu-sed" => :build # https://lists.gnu.org/archive/html/octave-maintainers/2016-09/msg00193.html
+  depends_on "libtool" => :build # for the patches
```
Comment (apjanke) [inline Formula/o/octave.rb:42]: Thanks for fixing this Qt 6.7 problem!

Out of curiosity: 

image

This addition of `depends_on "libtool" => :build` seems redundant with the existing plain (non-`:build`) `depends_on "libtool"` further down. The main (not `on_linux`) depends_on list now looks like this to me:

```
  depends_on "libtool" => :build # for the patches
[...]
  depends_on "libtool"
```

Does the `:build` version of the depends_on do something special here? I thought that all the regular depends_on dependencies were also installed/available at build time, and the `:build` dependencies meant "only install this when doing a build of this formula, but don't install it when installing the package from a bottle.

Comment (fxcoudert) [reply Formula/o/octave.rb:42]: It's a mistake

**Pull request merged**